daprdocs/content/en/reference/api/distributed_lock_api.md

```diff
@@ -0,0 +1,150 @@
+---
+type: docs
+title: "Distributed Lock API reference"
+linkTitle: "Distributed Lock API"
+description: "Detailed documentation on the distributed lock API"
+weight: 660
+---
+
+## Lock
+
+This endpoint lets you acquire a lock.
+
+### HTTP Request
+
+```
+POST http://localhost:<daprPort>/v1.0-alpha1/lock/<storename>
+```
+
+#### URL Parameters
+
+Parameter | Description
+--------- | -----------
+`daprPort` | The Dapr port
+`storename` | The `metadata.name` field component file. Refer to the [component schema] ({{< ref component-schema.md>}})
+
+#### Query Parameters
+
+None
+
+### HTTP Response codes
+
+Code | Description
+---- | -----------
+200  | Request successful
+204  | Empty Response
+400  | Malformed request
+500  | Request failed
+
+### HTTP Request Body
+
+The lock endpoint receives the following JSON payload:
+
+```json
+{
+    "resourceId": "",
+    "lockOwner": "",
+    "expiryInSeconds": 0
+}
+```
+
+### HTTP Response Body
+
+The lock endpoint returns the following payload:
+
+```json
+{
+    "success": true
+}
+```
+
+### Examples
+
+```shell
+curl -X POST http://localhost:3500/v1.0-alpha/lock/redisStore \
+  -H "Content-Type: application/json" \
+  -d '{
+        "resourceId": "lock1",
+        "lockOwner": "vader",
+        "expiryInSeconds": 60
+      }'
+
+{
+    "success": "true"
+}
+```
+
+## Unlock
+
+This endpoint lets you unlock an existing lock.
+
+### HTTP Request
+
+```
+POST http://localhost:<daprPort>/v1.0-alpha1/unlock/<storename>
+```
+
+#### URL Parameters
+
+Parameter | Description
+--------- | -----------
+`daprPort` | The Dapr port
+`storename` | The `metadata.name` field component file. Refer to the [component schema] ({{< ref component-schema.md>}})
+
+#### Query Parameters
+
+None
+
+### HTTP Response codes
+
+Code | Description
+---- | -----------
+200  | Request successful
+204  | Empty Response
+400  | Malformed request
+500  | Request failed
+
+### HTTP Request Body
+
+The unlock endpoint receives the following JSON payload:
+
+```json
+{
+    "resourceId": "",
+    "lockOwner": ""
+}
+```
+
+### HTTP Response Body
+
+The unlock endpoint returns the following payload:
+
+```json
+{
+    "status": 0
+}
+```
+
+The `status` field contains the following response codes:
+
+Code | Description
+---- | -----------
+0  | Success
```

Comment:
Why does this need to return a int? This is difficult to then figure out what this means. Should really return a descriptive string. If this is  design issue, suggest raising a GH issue for v1.9

Reply:
Actually, it's very common in RESTful API design to return status codes (much like HTTP status codes) that map to a specific behavior that is otherwise verbose.